restli-server/src/test/java/com/linkedin/restli/internal/server/model/TestRestLiApiBuilder.java

```diff
@@ -198,4 +198,33 @@ public void testActionReturnType(Class<?> resourceClass, Class<?> expectedAction
       Assert.assertEquals(resourceMethodDescriptor.getActionReturnType(), expectedActionReturnType);
     }
   }
+
+  @DataProvider(name = "unstructuredFinderReturnTypeData")
+  private Object[][] unstructuredFinderReturnTypeData()
+  {
+    return new Object[][]
+        {
+            {fooBarFinderResource.class, "The return type must be a RecordTemplate"}
+        };
+  }
+
+  /**
+   * Ensures that when finder methods are processed, if the return type is not of a Record, then it will be warned.
+   * For instance, it should recognize that the "logical" return type for a method
+   * {@code Task<ActionResult<String>> doFoo();} is {@code String.class}.
+   *
+   * @param resourceClass resource used as an input
+   */
+  @Test(dataProvider = "unstructuredFinderReturnTypeData")
```

Comment:
Given that there's only one case being tested, it's acceptable to hard code the inputs rather than using a data provider. Caveat would be if this case will likely get extended. Your call.

Reply:
I will add more resources class to test since it addresses this concern and also provide more coverage